src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -104,6 +104,19 @@ public void onStepRemoved(StepRemovedEvent event) {
     }
 
     @Subscribe
+    public void onStepMoved(StepMovedEvent event) {
+        final Step step = event.getStep();
+
+        synchronized (this) {
+            final int oldIndex = this.steps.indexOf(step);
+            this.steps.remove(oldIndex);
+
+            final int newIndex = Math.min(Math.max(oldIndex + event.getDistance(), 0), this.steps.size());
+            this.steps.add(newIndex, step);
+        }
```

Comment:
Ah. Okay. That makes sense. Can you put a comment explaining that above the line.
It's not really all that clear from the code.